oarepo_vocabularies/authorities/providers/orcid_provider.py

```diff
@@ -0,0 +1,153 @@
+        params = params or {}
+        access_token = self.orcid_client.get_search_token_from_orcid()
+
+        page = params.get("page", 1)
+        page_size = params.get("page_size", 20)
+        start = (page - 1) * page_size
+
+        results = self.orcid_client.search(params.get("q", ""), method="edismax", start=start, rows=page_size, access_token=access_token)
+
+        total = results.get("num-found")
+
+        orcids = [result['orcid-identifier']['path'] for result in results.get("result")]
+
+        records = [self.orcid_client.get_record(access_token, orcid) for orcid in orcids]
+
+        items = [self.to_vocabulary_item(record) for record in records]
+
+        return items, total
+
+    
+
+    def get(self, identity, item_id, **kwargs):
+        if not idutils.is_orcid(item_id):
+            raise AssertionError(f"{item_id} is not a valid ORCID identifier.")
+        
+        access_token = self.orcid_client.get_search_token_from_orcid()
+        
+        orcid_id = idutils.normalize_orcid(item_id)
+
+        record = self.orcid_client.get_record(access_token, orcid_id)
+        
+        if record is None:
+            raise KeyError(f"ORCID record {item_id} not found.")
+        
+        return self.to_vocabulary_item(record)
+    
+    @staticmethod
+    def get_affiliations(activities_summary):
+        affiliations = []
+        
+        for key, activity in activities_summary.items():
+            if key in ORCIDProvider.IGNORED_AFFILIATIONS:
+                continue
+            
+            if key == "educations":
+                summary_type = "education-summary"
+                
+            if key == "employments":
+                summary_type = "employment-summary"
+                
+            for affiliation in activity.get(summary_type, []):
+                organization = affiliation.get("organization", {})
+                organization_name = organization.get("name", "")
+                try:
+                    organization_id = organization.get("disambiguated-organization", {}).get("disambiguated-organization-identifier", "")
+                    id_source = organization.get("disambiguated-organization", {}).get("disambiguation-source", "")
+                    new_affiliation = {"name": organization_name, "id": f"{id_source}:{organization_id}"}
+                except AttributeError:
+                    new_affiliation = {"name": organization_name, "id": "N/A"}
+                
+                if new_affiliation not in affiliations:
+                    affiliations.append(new_affiliation)
+        
+        return affiliations
+    
+    @staticmethod
+    def to_vocabulary_item(orcid_item):
+        orcid_id = idutils.normalize_orcid(orcid_item.get("orcid-identifier", {}).get("path", ""))
+
+        # Personal information
+        person = orcid_item.get("person", {})
+
+        if person.get("name", {}).get("family-name") is None or person.get("name", {}).get("given-names") is None:
```

Comment:
def dict_get(d, *args):
   ... iteratively get the path
   
   family_name = dict_get(person, "name", "family_name", "value")
   given_name = ...
   if not family_name or not given_name: retunr None
   ...